bode/bode/models/task_relation.py

```diff
@@ -0,0 +1,79 @@
+import uuid
+
+from enum import Enum
+
+from sqlalchemy.dialects.postgresql import ENUM, UUID
+
+from bode.app import db
+
+
+class RelationType(Enum):
+    Dependent = "DEPENDENT"
+    Interchangable = "INTERCHANGABLE"
+    Subtask = "SUBTASK"
+
+    @classmethod
+    def list(cls):
+        return list(map(lambda c: c.value, cls))
+
+
+SYMMETRIC_RELATIONS = [RelationType.Interchangable.value]
+
+
+class TaskRelation(db.Model):
+    __tablename__ = "tasks_relations"
+
+    id = db.Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
+    first_task_id = db.Column(UUID(as_uuid=True), db.ForeignKey("tasks.id"), nullable=False)
+    second_task_id = db.Column(UUID(as_uuid=True), db.ForeignKey("tasks.id"), nullable=False)
+    type = db.Column(ENUM(*RelationType.list(), name="task_relation_type"), nullable=False)
+
+    __table_args__ = (
+        db.Index(
+            "task_relations_index",
+            first_task_id,
+            second_task_id,
+            unique=True,
+        ),
+        db.CheckConstraint("first_task_id <> second_task_id", name="self_relation_check"),
+    )
+
+    def create(**relation_data):
+        relation = TaskRelation(**relation_data)
+
+        db.session.add(relation)
+        if relation_data["type"] in SYMMETRIC_RELATIONS:
+            symmetric_data = relation_data.copy()
+            symmetric_data["first_task_id"] = relation_data["second_task_id"]
+            symmetric_data["second_task_id"] = relation_data["first_task_id"]
+
+            print(symmetric_data)
```

Comment:
Should we print that? If the log should be here, wdyt of adding a logger utility so that the reviewers know if it's a debug or server `print`? It doesn't have to be created in this PR

Reply:
Nope. I'll add a flake plugin and ESlint rule to catch wild `print/console` statements. Not sure if we'll need much logging in this app, but if yes then ofc we should use some proper logging solution like python's built-in [logging module](https://docs.python.org/3/library/logging.html).